### PR TITLE
New example lib part #2

### PR DESCRIPTION
**Pull request by alexandraBara**
Added execute option to example lib:
./go_fish.py example --execute --session_id 1 --> runs 'rocminfo' and stores result in result col in job table.

Added README to /tuna/example to show how to use the example lib.

### CODE REVIEW

File: tuna/example/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+New library integration 
+=======================
+
+Historically, MITuna only supported MIOpen tuning. The tuning infrastructure has been updated
+to support a new library integration through minimal steps.
```
Comment (JehandadKhan) [inline tuna/example/README.md]:
```suggestion
An example of how to integrate external applications in Tuna to utilize the scaling features of Tuna
```

File: tuna/example/example_worker.py [hunk 2]
```diff
@@ -30,33 +30,50 @@
...
+    super().__init__(**kwargs)
+    self.set_db_tables()
+
+  def set_db_tables(self):
+    """Initialize tables"""
+    self.dbt = ExampleDBTables(session_id=self.session_id)
+
+  def close_job(self):
+    """mark a job complete"""
+    self.set_job_state('completed')
```
Comment (cderb) [inline tuna/example/example_worker.py]: Don't need this function.

File: tuna/example/example_worker.py [hunk 3]
```diff
@@ -30,33 +30,50 @@
...
+    cmd_output = None
+    try:
+      cmd_output = self.run_cmd()
+    except ValueError as verr:
+      self.logger.info(verr)
+      failed_job = True
 
     if failed_job:
-      self.set_job_state('errored', result=result_str)
+      self.set_job_state('errored', result='')
```
Comment (cderb) [inline tuna/example/example_worker.py]: perhaps the error string could be placed in the result here

File: tuna/example/load_job.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      'Insert jobs into MySQL db',
+      [TunaArgs.VERSION, TunaArgs.ARCH, TunaArgs.NUM_CU, TunaArgs.SESSION_ID])
+  parser.add_argument('-l',
+                      '--label',
+                      type=str,
+                      dest='label',
+                      required=True,
+                      help='Label to annontate the jobs.',
+                      default='new')
+  parser.add_argument('--fin_steps', dest='fin_steps', type=str, default=None)
```
Comment (cderb) [inline tuna/example/load_job.py]: Reference to fin. Maybe rename with generics.

File: tuna/miopen/tables.py [hunk 5]
```diff
@@ -35,9 +35,8 @@
 from tuna.miopen.miopen_tables import BNBenchmark
 from tuna.miopen.benchmark import Framework, Model
 from tuna.config_type import ConfigType
-from tuna.dbBase.sql_alchemy import DbSession
-from tuna.miopen.session import Session
 from tuna.tables_interface import DBTablesInterface
+from tuna.miopen.session import Session
```
Comment (cderb) [inline tuna/miopen/tables.py]: Lets put this next to the other imports from the tuna.miopen directory.

File: tuna/miopen/fin_class.py [hunk 6]
```diff
@@ -84,9 +84,15 @@ def __init__(self, **kwargs):
...
-    #call to set_db_tables in super must come after config_type is set
     super().__init__(**kwargs)
 
+    #add cache directories
+    self.envmt.append(
+        f"MIOPEN_USER_DB_PATH=/tmp/miopenpdb/thread-{self.gpu_id}/config/miopen"
+    )
+    self.envmt.append(
+        f"MIOPEN_CUSTOM_CACHE_DIR=/tmp/miopenpdb/thread-{self.gpu_id}/cache")
+
```
Comment (cderb) [inline tuna/miopen/fin_class.py:99]: Thanks for moving this.

**Review by cderb — CHANGES_REQUESTED** [review]

File: tuna/example/tables.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    #for pylint
+    self.job_table = None
+    self.session_table = SessionExample
+
+    self.set_tables()
+
+  def set_tables(self, sess_class=SessionExample):
+    """Set appropriate tables based on requirements"""
+    super().set_tables(sess_class)
+    self.job_table = Job
```
Comment (reidkwja) [inline tuna/example/tables.py:50]: We can bypass the pylint few public methods by using functional style for this method: 
https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/too-few-public-methods.html

File: tuna/tables_interface.py [hunk 8]
```diff
@@ -26,12 +26,12 @@
 ###############################################################################
 """Module that encapsulates the DB representation for a library"""
 
-from tuna.miopen.miopen_tables import JobMixin
+from tuna.dbBase.sql_alchemy import DbSession
 
 
 #pylint: disable=too-few-public-methods
```
Comment (reidkwja) [inline tuna/tables_interface.py:32]: We can do the same for this as well

**Review by reidkwja — CHANGES_REQUESTED** [review]

File: tuna/example/load_job.py (hunk 4 above)
Comment (alexandraBara) [reply tuna/example/load_job.py]: this is not needed, will remove

**Review by reidkwja — APPROVED** [review]

**Review by cderb — APPROVED** [review]

**Review by junliume — APPROVED** [review]

**Pull request merged**